File: src/tsBusinessLayer/dtos/BookDTO.ts
import { IBook } from '../interfaces/IBook';
import { AuthorDTO } from './AuthorDTO';
import { BookPriceDTO } from './BookPriceDTO';
import { GenreDTO } from './GenreDTO';
import { LanguageDTO } from './LanguageDTO';

export class BookDTO implements IBook {
  public id: number;
  public Title: String;
  public Description: String;
  public ReleaseDate: Date;
  public ImageSrc: String;
  public Authors: Array<AuthorDTO>;
  public Genres: Array<GenreDTO>;
  public Languages: Array<LanguageDTO>;
  public Prices: Array<BookPriceDTO>;
  public ImageBase64?: String | null;

  constructor(
    id: number,
    title: String,
    description: String,
    imageSrc: String,
    releaseDate: Date,
    authors: Array<AuthorDTO>,
    genres: Array<GenreDTO>,
    languages: Array<LanguageDTO>,
    bookPrices: Array<BookPriceDTO>,
    imageBase64?: String | null,
  ) {
    this.id = id;
    this.Title = title;
    this.Description = description;
    this.ImageSrc = imageSrc;
    this.ReleaseDate = releaseDate;
    this.Authors = authors;
    this.Genres = genres;
    this.Languages = languages;
    this.Prices = bookPrices;
    this.ImageBase64 = imageBase64;
  }
}

File: Berix-Library-API/Application/DTOs/Books/BookDTO.cs
using Application.DTOs.Authors;
using Application.DTOs.Genres;
using Application.DTOs.Languages;
using Application.DTOs.Orders;
using Domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Application.DTOs.Books
{
    public class BookDTO
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string ImageSrc { get; set; }
        public DateTime ReleaseDate { get; set; }
        public ICollection<GenreDTO> Genres { get; set; } = new List<GenreDTO>();
        public ICollection<AuthorDTO> Authors { get; set; } = new List<AuthorDTO>();
        public ICollection<LanguageDTO> Languages { get; set; } = new List<LanguageDTO>();
        virtual public ICollection<BookPriceDTO> Prices { get; set; } = new List<BookPriceDTO>();
        virtual public ICollection<OrderInvoiceDTO>? OrderInvoices { get; set; } = new List<OrderInvoiceDTO>();
        public byte[] Image { get; set; }
    }
}
